CMakeLists.txt

```diff
@@ -1,34 +1,28 @@
-cmake_minimum_required (VERSION 2.8.8)
+cmake_minimum_required(VERSION 3.12)
 
-if (CMAKE_VERSION VERSION_LESS "3.1")
-  add_definitions(-std=c++11)
-else()
-  set(CMAKE_CXX_STANDARD 11)
-endif()
+project(safe VERSION 0.2.0)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
```

Comment:
is this line necessary ? (the 7)

Reply:
To be honest, I'm not quite sure. The intention behind is to prevent the integration of compiler extensions and stick to pure standard C++. It wont do any harm, though a usage of extensions is very unlikely here.

If you have any preference, let me know and I'll update this accordingly.

Reply:
I'm totally fine with disabling extensions. The Travis is set to test on clang and gcc, any use of compiler-specific extensions is dangerous...